mkgu_packaging/dicarlo/marques/marques_stim_common.py

```diff
@@ -0,0 +1,243 @@
+
+    stimuli['image_file_name'] = pd.Series(image_names)
+    stimuli['image_current_local_file_path'] = pd.Series(save_dir + os.sep + image_names)
+
+    stimuli.to_csv(save_dir + os.sep + 'stimulus_set', index=False)
+
+
+def gen_grating_stim(degrees, size_px, stim_name, grat_contrast, grat_pos, grat_rad, grat_sf, grat_orientation,
+                     grat_phase, save_dir):
+    if not (os.path.isdir(save_dir)):
+        os.mkdir(save_dir)
+
+    width = degrees
+
+    nStim = len(grat_pos) * len(grat_pos) * len(grat_contrast) * len(grat_rad) * len(grat_sf) * len(grat_orientation) \
+            * len(grat_phase)
+
+    print('Generating stimulus: #', nStim)
+
+    stimuli = pd.DataFrame({'image_id': [str(n) for n in range(nStim)], 'degrees': [width] * nStim})
+
+    image_names = nStim * [None]
+    image_local_file_path = nStim * [None]
+    all_y = nStim * [None]
+    all_x = nStim * [None]
+    all_c = nStim * [None]
+    all_r = nStim * [None]
+    all_s = nStim * [None]
+    all_o = nStim * [None]
+    all_p = nStim * [None]
+
+    i = 0
+    for y in np.arange(len(grat_pos)):
+        for x in np.arange(len(grat_pos)):
+            for c in np.arange(len(grat_contrast)):
+                for r in np.arange(len(grat_rad)):
+                    for s in np.arange(len(grat_sf)):
+                        for o in np.arange(len(grat_orientation)):
+                            for p in np.arange(len(grat_phase)):
+                                grat = Grating(width=width, pos=[grat_pos[y], grat_pos[x]],
+                                               contrast=grat_contrast[c], rad=grat_rad[r],
+                                               sf=grat_sf[s], orientation=grat_orientation[o],
+                                               phase=grat_phase[p],
+                                               stim_id=np.uint64(
+                                                   y * 10e9 + x * 10e7 + r * 10e5 + s * 10e3 + o * 10e1 + p),
+                                               format_id='{0:012d}', save_dir=save_dir, size_px=[size_px, size_px],
+                                               type_name=stim_name)
+                                image_names[i] = (grat.save_stimulus())
+                                image_local_file_path[i] = save_dir + os.sep + image_names[i]
+                                all_y[i] = grat_pos[y]
+                                all_x[i] = grat_pos[x]
+                                all_c[i] = grat_contrast[c]
+                                all_r[i] = grat_rad[r]
+                                all_s[i] = grat_sf[s]
+                                all_o[i] = grat_orientation[o]
+                                all_p[i] = grat_phase[p]
+                                i += 1
+
+    stimuli['pos_y'] = pd.Series(all_y)
+    stimuli['pos_x'] = pd.Series(all_x)
+    stimuli['contrast'] = pd.Series(all_c)
+    stimuli['radius'] = pd.Series(all_r)
+    stimuli['sf'] = pd.Series(all_s)
```

Comment:
as above, please write this out to avoid confusion

Reply:
Done